inference-engine/src/mkldnn_plugin/nodes/mkldnn_color_convert_node.h

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <mkldnn_node.h>
+#include <utils/multidim_map.hpp>
+#include <functional>
+#include <tuple>
+#include <array>
+
+namespace MKLDNNPlugin {
+
+class MKLDNNColorConvertNode : public MKLDNNNode {
+public:
+    MKLDNNColorConvertNode(const std::shared_ptr<ngraph::Node>& op,
+                           const mkldnn::engine& eng,
+                           MKLDNNWeightsSharing::Ptr &cache);
+
+public:
+    class Converter {
+    public:
+        using PrimitiveDescs = std::vector<std::tuple<std::vector<PortConfigurator>,
+                                                       std::vector<PortConfigurator>,
+                                                       impl_desc_type,
+                                                       bool>>;
+        using Shapes = std::vector<VectorDims>;
+
+        static constexpr size_t N_DIM = 0;
+        static constexpr size_t H_DIM = 1;
+        static constexpr size_t W_DIM = 2;
+        static constexpr size_t C_DIM = 3;
+
+        using ColorFormat = std::array<uint8_t, 3>;
+
+        Converter(MKLDNNNode *node, const ColorFormat & colorFormat);
+        virtual ~Converter() = default;
+        InferenceEngine::Precision inputPrecision(size_t idx) const;
+        InferenceEngine::Precision outputPrecision(size_t idx) const;
+        const void * input(size_t idx) const;
+        void * output(size_t idx) const;
+        const VectorDims & inputDims(size_t idx) const;
+        virtual Shapes shapeInfer() const = 0;
+        virtual void prepare() = 0;
+        virtual void execute(mkldnn::stream strm) = 0;
+
+    protected:
+        MKLDNNNode *_node;
+        ColorFormat _colorFormat;   // RGB: {0,1,2}, BGR: {2,1,0}
+    };
+
+private:
```

Comment:
public?
I guess the only reason why it is working is that MKLDNNGraph is a friend.

Reply:
It works, because in MKLDNNNode they defined as public.

Reply:
Done, defined as public